Shared/Samples/Augment reality to navigate route/AugmentRealityToNavigateRouteView.ARSceneView.swift

```diff
@@ -0,0 +1,275 @@
+        
+        /// The route task that solves the route using the online routing service, using API key authentication.
+        let routeTask = RouteTask(url: URL(string: "https://route-api.arcgis.com/arcgis/rest/services/World/Route/NAServer/Route_World")!)
+        
+        /// The parameters for route task to solve a route.
+        var routeParameters = RouteParameters()
+        
+        /// The data source to track device location and provide updates to route tracker.
+        let locationDataSource = SystemLocationDataSource()
+        
+        /// The status text displayed to the user.
+        @Published var statusText = ""
+        
+        init() {
+            elevationSurface.addElevationSource(elevationSource)
+            scene.baseSurface = elevationSurface
+        }
+        
+        /// Loads the scene elevation source.
+        func loadElevationSource() async throws {
+            await scene.baseSurface.elevationSources.load()
+        }
+        
+        /// Tracks the location datasource locations.
+        func startTrackingLocation() async throws {
+            for await location in locationDataSource.locations {
+                try await routeTracker?.track(location)
+            }
+        }
+        
+        /// Creates a graphics overlay and adds a graphic (with solid yellow 3D tube symbol)
+        /// to represent the route.
+        func makeRouteOverlay() async throws {
+            let graphicsOverlay = GraphicsOverlay()
+            graphicsOverlay.sceneProperties.surfacePlacement = .absolute
+            let strokeSymbolLayer = SolidStrokeSymbolLayer(
+                width: 1.0,
+                color: .yellow,
+                lineStyle3D: .tube
+            )
+            let polylineSymbol = MultilayerPolylineSymbol(symbolLayers: [strokeSymbolLayer])
+            let polylineRenderer = SimpleRenderer(symbol: polylineSymbol)
+            graphicsOverlay.renderer = polylineRenderer
+            
+            if let routeResult,
+               let originalPolyline = routeResult.routes.first?.geometry,
+               let elevatedPolyline = try await addingElevation(3, to: originalPolyline) {
+                let routeGraphic = Graphic(geometry: elevatedPolyline)
+                graphicsOverlay.addGraphic(routeGraphic)
+            }
+            
+            routeGraphicsOverlay = graphicsOverlay
+        }
+        
+        /// Densifies the polyline geometry so the elevation can be adjusted every 0.3 meters and adds
+        /// an elevation to the geometry.
+        /// - Parameters:
+        ///   - z: The z elevation.
+        ///   - polyline: The polyline geometry of the route.
+        /// - Returns: A polyline with adjusted elevation.
+        private func addingElevation(_ z: Double, to polyline: Polyline) async throws -> Polyline? {
+            if let densifiedPolyline = GeometryEngine.densify(polyline, maxSegmentLength: 0.3) as? Polyline {
```

Comment:
This if-let seems like it would be better as a `guard`, but I'm good either way. Take it or leave it.